anomaly-detection/src/main/java/io/pravega/anomalydetection/event/pipeline/PravegaAnomalyDetectionProcessor.java

```diff
@@ -0,0 +1,183 @@
+/**
+ *
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries.
+ *
+ */
+package io.pravega.anomalydetection.event.pipeline;
+
+import io.pravega.anomalydetection.event.AppConfiguration;
+import io.pravega.anomalydetection.event.serialization.PravegaDeserializationSchema;
+import io.pravega.anomalydetection.event.state.Event;
+import io.pravega.anomalydetection.event.state.Result;
+import io.pravega.client.stream.impl.JavaSerializer;
+import io.pravega.connectors.flink.FlinkPravegaReader;
+import io.pravega.shaded.com.google.gson.Gson;
+import org.apache.flink.api.common.functions.FoldFunction;
+import org.apache.flink.api.common.functions.RuntimeContext;
+import org.apache.flink.runtime.state.filesystem.FsStateBackend;
+import org.apache.flink.shaded.org.apache.curator.shaded.com.google.common.collect.Sets;
+import org.apache.flink.streaming.api.CheckpointingMode;
+import org.apache.flink.streaming.api.TimeCharacteristic;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.apache.flink.streaming.api.functions.timestamps.BoundedOutOfOrdernessTimestampExtractor;
+import org.apache.flink.streaming.api.windowing.assigners.TumblingEventTimeWindows;
+import org.apache.flink.streaming.api.windowing.time.Time;
+import org.apache.flink.streaming.connectors.elasticsearch.ElasticsearchSinkFunction;
+import org.apache.flink.streaming.connectors.elasticsearch.RequestIndexer;
+import org.apache.flink.streaming.connectors.elasticsearch5.ElasticsearchSink;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.client.Requests;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.time.Instant;
+import java.util.*;
+
+public class PravegaAnomalyDetectionProcessor implements IPipeline {
+
+	@Override
+	public void run(AppConfiguration appConfiguration) throws Exception {
```

Comment:
To work best with savepoints, we really should set the `uid(...)` on all operators.